go/vt/vtgate/evalengine/eval_result.go

```diff
@@ -32,9 +32,29 @@ import (
 	"vitess.io/vitess/go/vt/vtgate/evalengine/decimal"
 )
 
+type flag uint16
+
 const (
-	flagHex = 1 << iota
-	flagBit
+	// flagNull marks that this value is null; implies flagNullable
```

Comment:
This information is awesome. We could extend it, and hook it up with the schema tracker so we know if columns are nullable.

I know at least one place where I would like to use this information in the planner, so we should consider making it a concept that is readable from other packages as well

Reply:
Noted, I'll add that to my TODO list.